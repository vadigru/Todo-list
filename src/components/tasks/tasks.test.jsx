import React from "react";
import renderer from "react-test-renderer";

import Tasks from "./tasks";

it(`Should render Tasks component`, () => {
  const tasks = [
    {
      task: `Task 1`,
      isActive: true
    },
    {
      task: `Task 2`,
      isActive: true
    },
  ];

  const tree = renderer
    .create(
        <Tasks
          tasks={tasks}
          onTaskClick={() => {}}
          onDeleteButtonClick={() => {}}
          onDropUpdate={() => {}}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
