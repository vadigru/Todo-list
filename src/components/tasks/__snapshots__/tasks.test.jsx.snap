// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Tasks component 1`] = `
<ul
  className="tasks"
  data-rbd-droppable-context-id="0"
  data-rbd-droppable-id="dp1"
>
  <li
    aria-describedby="rbd-hidden-text-0-hidden-text-0"
    className="tasks__item "
    data-rbd-drag-handle-context-id="0"
    data-rbd-drag-handle-draggable-id="0"
    data-rbd-draggable-context-id="0"
    data-rbd-draggable-id="0"
    draggable={false}
    onDragStart={[Function]}
    onTransitionEnd={null}
    role="button"
    style={
      Object {
        "transform": null,
        "transition": null,
      }
    }
    tabIndex={0}
  >
    <button
      className="tasks__check"
      onClick={[Function]}
    >
      <img
        alt="unchecked"
        src="img/checkbox-unchecked.svg"
      />
    </button>
    <span
      className="tasks__title"
    >
      Task 1
    </span>
    <button
      className="tasks__delete"
      onClick={[Function]}
    >
      <img
        alt="delete"
        src="img/icons8-delete.svg"
      />
    </button>
  </li>
  <li
    aria-describedby="rbd-hidden-text-0-hidden-text-0"
    className="tasks__item "
    data-rbd-drag-handle-context-id="0"
    data-rbd-drag-handle-draggable-id="1"
    data-rbd-draggable-context-id="0"
    data-rbd-draggable-id="1"
    draggable={false}
    onDragStart={[Function]}
    onTransitionEnd={null}
    role="button"
    style={
      Object {
        "transform": null,
        "transition": null,
      }
    }
    tabIndex={0}
  >
    <button
      className="tasks__check"
      onClick={[Function]}
    >
      <img
        alt="unchecked"
        src="img/checkbox-unchecked.svg"
      />
    </button>
    <span
      className="tasks__title"
    >
      Task 2
    </span>
    <button
      className="tasks__delete"
      onClick={[Function]}
    >
      <img
        alt="delete"
        src="img/icons8-delete.svg"
      />
    </button>
  </li>
</ul>
`;
